src/map/script.cpp

```diff
@@ -24794,6 +24794,36 @@ BUILDIN_FUNC(hateffect){
 	return SCRIPT_CMD_SUCCESS;
 }
 
+/**
+ * Check if hat effect is enabled on player
+ * *has_hateffect(<effect_id>{,<char_id>});
+ */
+BUILDIN_FUNC(has_hateffect) {
+#if PACKETVER_MAIN_NUM >= 20150507 || PACKETVER_RE_NUM >= 20150429 || defined(PACKETVER_ZERO)
+	map_session_data* sd;
+
+	if (!script_charid2sd(3, sd))
+		return SCRIPT_CMD_FAILURE;
```

Comment:
SCRIPT_CMD_FAILURE doesn't stop the script. 0 must be pushed into the stack

Reply:
Do we want it to stop? Also, does `script_pushint(st, 0)` stop it. I thought it's `st->state = END;`